readthedocs/search/tasks.py

```diff
@@ -118,3 +122,51 @@ def index_missing_objects(app_label, model_name, document_class, index_generatio
     log.info("Indexed %s missing objects from model: %s'", queryset.count(), model.__name__)
 
     # TODO: Figure out how to remove the objects from ES index that has been deleted
+
+
+@app.task(queue='web')
+def delete_old_search_queries_from_db():
+    """
+    Delete old SearchQuery objects.
+
+    This is run by celery beat every day.
+    """
+    last_3_months = timezone.now().date() - timezone.timedelta(days=90)
+    search_queries_qs = SearchQuery.objects.filter(
+        created__date__lte=last_3_months,
+    )
+
+    if search_queries_qs.exists():
+        log.info('Deleting search queries for last 3 months. Total: %s', search_queries_qs.count())
+        search_queries_qs.delete()
+
+
+@app.task(queue='web')
+def record_search_query(project_slug, version_slug, query, total_results):
```

Comment:
Why are we passing `total_results` then not saving it?

Reply:
This will also cause us to not store search queries with 0 results, which feels wrong.

Reply:
My motive was to not save the query if no results.
If we save every queries, we might end up having queries which don't show anything.
Like, with the readthedocs-sphinx-search, a search query is made after some time the user stops typing (~500ms).
So, for typing -- 'advertising',
we might end up having multiple queries: `adve`, `advertis`, `advertising`.